s3_log_shipper/shipper.py

```diff
@@ -0,0 +1,72 @@
+import codecs
+import gzip
+import json
+import logging
+from typing import Tuple, Optional, Any, Dict
+
+from botocore.client import BaseClient
+from botocore.response import StreamingBody
+from redis import StrictRedis
+
+from s3_log_shipper.parsers import ParserManager, Parser
+
+log: logging.Logger = logging.getLogger(__name__)
+
+
+class RedisLogShipper:
+    def __init__(
+        self,
+        redis_endpoint: StrictRedis,
+        parser_manager: ParserManager,
+        s3_client: BaseClient,
+    ):
+        self.redis_endpoint: StrictRedis = redis_endpoint
+        self.parser_manager: ParserManager = parser_manager
+        self.s3_client: BaseClient = s3_client
+
+    def ship(self, bucket: str, key: str):
+        maybe_parser: Optional[Tuple[Parser, dict]] = self.parser_manager.get_parser(
```

Comment:
Based on line #49, it looks like this should be an Optional[Tuple[Parser, Optional[dict]]]?

Reply:
Afraid I don't understand what you're getting at here - `get_parser` returns `Optional[Tuple[Parser, dict]]`, and line 49 is:

```
log.error(f"Couldn't grok log line {line}")
```

Which is unrelated?